docs/http-challenge/s3.md

```diff
@@ -0,0 +1,72 @@
+# Variables for s3 http-challenge
+
+| Variable                            | Required | Default   | Description
+|-------------------------------------|----------|-----------|------------
+| letsencrypt_s3_bucket_name          | yes      |           | name of the s3 bucket which should be used
+| letsencrypt_s3_access_key           | yes      |           | aws access key for API user of s3 bucket
+| letsencrypt_s3_secret_key           | yes      |           | aws secret key for API user of s3 bucket
+| letsencrypt_s3_config_region        | no       | us-west-1 | aws s3 region in which bucket can be found
+
+## Validation
+
+You have to set a redirect rule in your proxy or webserver to allow the acme challenge bot to read the file, during the http-01 challenge to work:
+
+**HaProxy(Version <=> 1.5):**
+
+```bash
+  # lets encrypt redirect
+  http-request del-header X-REWRITE
+  http-request add-header X-REWRITE %[url] if { path_beg /.well-known/acme-challenge }
+  http-request replace-header X-REWRITE ^(.well-known/acme-challenge/.*)?$ /\1 if { hdr_cnt(X-REWRITE) gt 0 }
+  http-request redirect code 301 location https://letsencrypt-challenge-bucket.s3.amazonaws.com%[hdr(X-REWRITE)] if { hdr_cnt(X-REWRITE) gt 0 }
```

Comment:
should we somehow highlight, that `https://letsencrypt-challenge-bucket.s3.amazonaws.com´ is a variable / must be changed by the user?

Reply:
Hm, maybe you are right. I will add some kind of hint.

Reply:
Added hint and changed URL naming